src/Microsoft.AspNetCore.Mvc.Core/UrlHelperExtensions.cs

```diff
@@ -471,6 +471,12 @@ public static string Page(this IUrlHelper urlHelper, string pageName, object val
                 routeValues["handler"] = pageHandler;
             }
 
+            if (!routeValues.ContainsKey("area") &&
+                ambientValues.TryGetValue("area", out var area))
+            {
+                routeValues["area"] = area;
+            }
+
```

Comment:
This shouldn't be here, we don't special case `area` like we do for `[controller, action]` and `[page, handler]`

Reply:
See https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs